### PR TITLE
commentchar

### PR DESCRIPTION
**Pull request by davidanthoff**
This finishes the work in #71 and should be ready for a merge.

### CODE REVIEW

File: src/util.jl [hunk 1]
```diff
@@ -214,6 +214,21 @@ function eatnewlines(str, i=1, l=lastindex(str))
...
 
+# Move past consecutive lines that start with commentchar.
+# Return a tuple of the new pos in str and the amount of comment lines moved past.
+function eatcommentlines(str, i=1, l=lastindex(str), commentchar::Union{Char, Nothing}=nothing) 
+    commentchar === nothing && return i, 0
+
+    count = 0
+    while i <= l && str[i] == commentchar
+        i = getlineend(str, i)
+        _, i = iterate(str, i)
```
Comment (shashi) [inline src/util.jl]: check if end of file before this?

**Comment by shashi** [comment]
except for that one comment, I approve of this. :+1:  Great feature!

File: src/util.jl (hunk 1 above)
Comment (davidanthoff) [reply src/util.jl]: I now check the return value for `nothing`, that branch should also generally improve performance.

**Comment by davidanthoff** [comment]
Thanks @nicholasphair for getting this started!

**Pull request merged**